### PR TITLE
VB Overload Resolution Priority - handle operators and respect language version

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @cston, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs [hunk 1]
```diff
@@ -2706,4 +2706,94 @@ static void Main()
...
+
+    static void M1(long x) => System.Console.Write(3);
+}
+""";
+
+        CompileAndVerify([source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "3").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void LiftedOperator_01()
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs:2768]: Is this meaningfully different than `UnaryOperator_Allowed`?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs:2768]: Probably not, but it is clear what this test is about and, therefore, it is less likely that the test will be modified in a way that the lifted/un-lifted aspect is not covered

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb [hunk 2]
```diff
@@ -983,5 +1257,114 @@ BC30526: Property 'M' is 'ReadOnly'.
...
+        Public Sub ConversionOperator_01()
+            Dim compilationDef = "
+Structure S
+
+    public shared widening operator CType(x As Integer) As S
+        System.Console.Write(1)
+        return Nothing
+    End Operator
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(1)>
```
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb:1350]: > Are we planning to report an error here? The equivalent C# is disallowed.

There is an open design question about that in the speclet

**Review by 333fred — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review

**Review by cston — APPROVED** [review]

**Pull request merged**